courseware/api.py

```diff
@@ -816,3 +834,26 @@ def delete_oauth_application():
         name=settings.OPENEDX_OAUTH_APP_NAME
     ).delete()
     return _, deleted_applications_count
+
+
+def validate_name_with_edx(name):
```

Comment:
We don't have an `openedx` directory in xPro, but it exists in MITx Online in replace of `courseware` for handling all API interactions with the edX platform. This is beneficial as it centralizes interactions with Open edX.

In xPro, courseware serves a similar role e.g. both course-related and user-specific Open edX API interactions.
The name courseware can be misleading since it implies only course-specific API interactions, while it actually includes other APIs interactions. I would recommend to lets keep it here to have common directory to interact with openedx similar to what we have in mitxonline and should have some meaningful name in future that serves the purpose to interact with openedx.

Reply:
Alright, I agree. Let's keep it here for now. But my recommendation would be to create and openedx app and move this there but we can do that later in a refactoring PR.